### ISSUE TITLE
Paypal subscriptions aren't cancellable

### ISSUE THREAD

**Issue by kbourgoin** [kbourgoin/hiicart#1]
The support for this in the current is quite poor.  HiiCart needs to implement Website Payments Pro, which will allow greater control over subscription terms.

https://cms.paypal.com/us/cgi-bin/?cmd=_render-content&content_ID=developer/howto_api_reference

https://cms.paypal.com/us/cgi-bin/?cmd=_render-content&content_ID=developer/e_howto_api_WPWebsitePaymentsPro#id0d434473-8817-4a4e-822a-bb23730a17e7

**Comment by kbourgoin** [comment]
I've started work on a paypal2 gateway to handle this.  If I can't finish it soon, I'll push the branch to github in case others want to work on it.

**Comment by kbourgoin** [comment]
I wasn't able to finish this in time, so I uploaded what is done.  Payments Pro works a lot like Amazon Payments.  The user needs to authorize the payment with them, and then we get a token to use and to make a Pay request.  The big difference is that paypal needs an interstital page with a 'receipt' on it before we make the pay call.

The best way to do this will be to have RETURN_URL point to something outside HiiCart showing the receipt.  That will have a form which submits back to HiiCart, which will then issue the pay request, or set up the recurring payment.  This should allow customization of the receipt page while keeping the core functionality of HiiCart separate from user-facing pages.

I've updated the example app to be functional enough to develop this.  It's ugly as heck, but should be faster than doing it within the integrating app.

I might have some time in the next week or two to work on this, but I'm putting the code out here in case I don't.
